src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -664,8 +735,20 @@ zocl_xclbin_read_axlf(struct drm_zocl_dev *zdev, struct drm_zocl_axlf *axlf_obj)
 
 	zocl_free_sections(zdev);
 
+	if (zocl_check_sect(PARTITION_METADATA, axlf) &&
```

Comment:
check if you can use xrt_xclbin_get_section_num instead of writing a new api

Reply:
yes but xrt_xclbin_get_section_num takes other argument like offset and size so encapsulation these in api can be a good option